### PR TITLE
Interfaces for two phase pyrolysis and bug fix

### PR DESCRIPTION
**Pull request by hwpang**
Added interfaces for two-phase pyrolysis and fixed some bugs for four party reactions and multi-domain reactor.

### CODE REVIEW

File: src/Domain.jl [hunk 1]
```diff
@@ -1503,6 +1503,14 @@ end
             dydt[d.indexes[3]] -= inter.Vout(t)
         end
     end
+    for inter in interfaces
+        if isa(inter,ConstantVOutlet) && d == inter.domain #ConstantVOutlet has to be the last to be evaluated to work correctly
```
Comment (mjohnson541) [inline src/Domain.jl:1507]: It doesn't actually have to be evaluated last. It just has to be evaluated after dVdt has been modified by anything else. This distinction is super important for anyone adding a similar interface such as one that adds gas to keep the pressure constant.

**Review by mjohnson541 — COMMENTED** [review]
Great work! Number of comments. Also can we add an IJulia notebook example?

File: src/Interface.jl [hunk 2]
```diff
@@ -164,7 +164,7 @@ function getstoichmatrix(domain1,domain2,rxns)
 end
 
 function getinterfacereactioninds(domain1,domain2,reactions)
-    indices = zeros(Int64,(6,length(reactions)))
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Interface.jl:167]: Yep!

File: src/Reactor.jl [hunk 3]
```diff
@@ -102,14 +102,16 @@ function Reactor(domains::T,y0s::W1,tspan::W2,interfaces::Z=Tuple(),ps::X=DiffEq
...
         domain.indexes[1] = k
         k += Nspcs
         domain.indexes[2] = k-1
         y0[domain.indexes[1]:domain.indexes[2]] = y0s[j][1:Nspcs]
         for m = 3:2+Ntherm
+            for (therm,ind) in domain.thermovariabledict
+                if ind == domain.indexes[m]
+                    domain.thermovariabledict[therm] = n
+                end
+            end
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Reactor.jl:114]: The thermo variables in the multi-domain simulation are compiled to the end of the y vector from the end towards the head of the vector, which is not necessarily k-1 space away from its original position. To make sure the thermo variable dict is updated correctly, we need to update it while we are filling the thermo variable to their new position.

I will add these to the commit message!

File: src/Interface.jl [hunk 4]
```diff
@@ -303,3 +303,65 @@ struct VolumetricFlowRateOutlet{V,F1<:Function} <: AbstractBoundaryInterface
     Vout::F1
 end
 export VolumetricFlowRateOutlet
+
+struct VaporLiquidMassTransferInternalInterfaceConstantT{D1,D2,B} <: AbstractInternalInterface
+    domain1::D1
+    domain2::D2
+    masstransferspcnames::Array{String,1}
+    masstransferarray::B
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Interface.jl:322]: Good point

File: src/Domain.jl (hunk 1 above)
Comment (hwpang) [reply src/Domain.jl]: Yep you're right

File: src/Interface.jl [hunk 5]
```diff
@@ -304,6 +304,17 @@ struct VolumetricFlowRateOutlet{V,F1<:Function} <: AbstractBoundaryInterface
 end
 export VolumetricFlowRateOutlet
 
+"""
+ConstantVOutlet is designed for gas phase domain such that the flow rate of this outlet will adjust to maintain the volume of the 
+    domain to be constant. This is particularly useful to simulate any vapor-liquid phase system where the gas phase outlet
+    is determined by the amount of evaporation.
+"""
+struct ConstantVOutlet{V} <: AbstractBoundaryInterface
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Interface.jl]: Solved

**Comment by hwpang** [comment]
@mjohnson541 I think I addressed all the comments and all tests have passed.

**Review by mjohnson541 — APPROVED** [review]

**Pull request merged**